File: beautiful-array/beautiful-array.js
var beautifulArray = function(n) {
    arr = [1];
    while(arr.length < n) {
        let tmp = [];
        arr.forEach(x=> {
            if(x*2-1 <= n) {
                tmp.push(x*2-1);
            }
        })
        arr.forEach(x =>{
             if(x*2 <= n) {
                tmp.push(x*2);
            }
        })
        arr = tmp;
    }
    return arr
};

File: beautiful-array/beautiful-array.py
class Solution:
    def beautifulArray(self, n: int) -> List[int]:
        arr = [1]
        while len(arr) < n: 
            arr = [2*x-1 for x in arr] + [2*x for x in arr]
        return [x for x in arr if x <= n]